users/service.go

```diff
@@ -694,6 +694,19 @@ func (svc service) addClientPolicyRollback(ctx context.Context, userID string, r
 }
 
 func (svc service) updateClientPolicy(ctx context.Context, userID string, role mgclients.Role) error {
+	res, err := svc.auth.Authorize(ctx, &magistrala.AuthorizeReq{
```

Comment:
This authorization should happen once not during rolling back. This is because when we roll back we are sure the user is in the platform

```go
client := mgclients.Client{
	ID:        cli.ID,
	Role:      cli.Role,
	UpdatedAt: time.Now(),
	UpdatedBy: tokenUserID,
}

if _, err := svc.authorize(ctx, auth.UserType, auth.UsersKind, tokenUserID, auth.MembershipPermission, auth.PlatformType, auth.MagistralaObject); err != nil {
	return mgclients.Client{}, err
}

if err := svc.updateClientPolicy(ctx, cli.ID, cli.Role); err != nil {
	return mgclients.Client{}, err
}
```

Reply:
@rodneyosodo, since we want to verify if the user whose role the super admin wants to update is on the platform, I think the subject should be `client.ID`.

Reply:
Update accordingly

Reply:
Okay